challenge.c

```diff
@@ -50,10 +54,14 @@ uint16_t q3(uint8_t x, uint8_t y) {
 // Note: The array contains 8-bit unsigned integers.
 //-------------------------------------------------------------------------
 int32_t q4(uint8_t * array, uint32_t arrayLength) {
-    for (uint8_t i = 0; i <= arrayLength; i++) {
-        int32_t sum = 0;
+    if (!array) {
+        return -1;
+    }
+    int32_t sum = 0;
+    for (uint8_t i = 0; i < arrayLength; i++) {
```

Comment:
Your tests were stuck in an infinite loop because of an error on this line. It's subtle, but try and find it. Hint: What's the max value `i` can go to here?

Reply:
Oh I see sorry my bad.